internal/controllers/server/reconcile.go

```diff
@@ -40,6 +40,9 @@ import (
 // +kubebuilder:rbac:groups=openstack.k-orc.cloud,resources=servers,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=openstack.k-orc.cloud,resources=servers/status,verbs=get;update;patch
 
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
```

Comment:
We might want to move this to `internal/scope/provider.go` as it also needs access to secrets and is more generic than the server controller.

Reply:
See https://github.com/k-orc/openstack-resource-controller/pull/143

I think we can safely have both, though, as they both require secrets.